proposals/NNNN-reinstate-self-for-accessing-instance-properties-and-functions.md

```diff
@@ -0,0 +1,99 @@
+# Re-instate mandatory self for accessing instance properties and functions  
+
+* Proposal: [SE-NNNN](https://github.com/apple/swift-evolution/blob/master/proposals/NNNN-mandatory-self-for-accessing-instance-properties-and-functions.md)
+* Author(s): [David Hart](https://github.com/hartbit)
+* Status: **Review**
+* Review manager: TBD
+
+## Introduction
+
+Swift used to force using `self.` when accessing instance properties and functions, but since dropped this requirement (only keeping it for closure captures). It now feels inconsistent, and we have lost the automatic documentation of instance properties vs local variables and instance functions vs local functions or closures. This proposal offers to re-instate the original behaviour.
```

Comment:
That was only a requirement for referencing the instance from within local functions, which are, in almost all regards, closures in swift. You are suggesting code within class methods (not local functions) use self to reference instance properties/methods, are you not? I made a gist to illustrate the differences listed above. https://gist.github.com/aceysmith/52548b084ec798bcf783